src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1775,7 +1810,8 @@ AddTypeConversion(Node *originalAggregate, Node *newExpression)
  */
```

Comment:
We should also extend the comment here for count distinct on non-distribution columns "treatment for count distinct operator if it is used in repartition subqueries"

Reply:
done